### ISSUE TITLE
Weird error when trying to connect to Cassandra database

### ISSUE THREAD

**Issue by SeanOMik** [scylladb/scylla-rust-driver#272]
When I try to connect to my Cassandra database, it gives me a confusing error:
```
Failed to create connection to cassandra: DbError(ServerError, "java.lang.IndexOutOfBoundsException: index: 4, length: 335569761 (expected: range(0, 21))")
```
Here's my code:
```rs
let cassandra_host = env::var("CASSANDRA_HOST").expect("Failed to find cassandra host! (Define env variable: CASSANDRA_HOST)");
let cassandra_port = env::var("CASSANDRA_PORT").expect("Failed to find cassandra port! (Define env variable: CASSANDRA_PORT)");
let cassandra_uri = format!("{}:{}", cassandra_host, cassandra_port);

let cassandra_client: Arc<Session> = Arc::new(
    SessionBuilder::new()
    .known_node(cassandra_uri)
    .user("cassandra", "cassandra")
    .build()
    .await
    .expect("Failed to connect to Cassandra"));
```

I tried to set the URI manually (without the env stuff) and it still happens. 

The odd thing though, is that if I remove the `.user` part of the builder (so removing auth), I get a different error:
```
Failed to create connection to cassandra: ProtocolError("Error parsing message frame")
```

**Comment by psarna** [comment]
Please provide more context to help us reproduce the issue:
1. What release of the driver are you trying out? 0.1.0? 0.2.0? built from newest source?
2. What's your exact Cassandra release? I just tested your code and it worked just fine for both 3.11.9 and 4.0-rc2
3. What values do you pass as your `CASSANDRA_HOST` and `CASSANDRA_PORT` parameters? I tested "localhost" + "9042" on my local setup, worked just fine.

**Comment by SeanOMik** [comment]
I was trying 0.2.0 of the driver.
I think the Cassandra version is 3.11.10 going off of the [docker image version](https://hub.docker.com/r/bitnami/cassandra/tags?page=1&ordering=last_updated) (latest) I'm using.

For the environmental variables, I tried "powersipper" + "9042". "powersipper" is the local network machine that runs the Cassandra DB. I also tried the actual IP instead of the hostname: "192.168.87.139" + "9042". I also tried to just set `cassandra_uri` manually to "powersipper:9042" and "192.168.87.139:9042"

**Comment by psarna** [comment]
Thanks - do you also have any specific custom Cassandra configuration, like authentication? Meanwhile, I'll try to reproduce the problem with their docker image.

**Comment by psarna** [comment]
Ok, confirmed, thanks. I suspect that this bitnami/cassandra container most likely uses an authentication mechanism that's not yet supported by our driver. I'll try to figure out what's the exact issue. I also compared with the official `cassandra:3.11.10` docker container and it works just fine, which supports my theory about custom configuration.

**Comment by psarna** [comment]
That's probably the message which our driver for some reason failed to parse:
![s-r-d-f](https://user-images.githubusercontent.com/10433047/126527585-4a1114e6-4d71-457a-bd23-91f8d829d670.png)
I don't know why yet, but I'll mark this issue as a legitimate bug and we'll look into it, thanks.

**Comment by SeanOMik** [comment]
Okay cool, thanks. In the meantime I might switch to the official image, or just stay with cdrs until the bug is fixed.

**Comment by piodul** [comment]
The part related to "Error parsing message frame" might be caused by the same problem as in https://github.com/scylladb/scylla-rust-driver/issues/243.

The `IndexOutOfBoundsException` must be something else.

**Comment by piodul** [comment]
I sent a PR which fixes the "IndexOutOfBoundsException" issue: https://github.com/scylladb/scylla-rust-driver/pull/273

**Comment by SeanOMik** [comment]
Awesome, just tested it with my project, and that or works great, thanks!

**Comment by psarna** [comment]
Fixed by https://github.com/scylladb/scylla-rust-driver/pull/273

**Comment by dragon-D** [comment]
I also had this error  `on an Err value: ProtocolError("Error parsing message frame")`
It appears like this. After running for a period of time, multiple requests are normal and sometimes errors, and the error returns Error parsing message frame

**Comment by psarna** [comment]
@dragon-D can you collect a report via wireshark or tcpdump? It would be interesting to see which requests caused the error. Also, please rerun your scenario on the main branch, where the authentication problem is already fixed? (By specifying git as the dependency source in Cargo.toml: `scylla = { git = "https://github.com/scylladb/scylla-rust-driver" }`)

**Comment by psarna** [comment]
edit: I released `0.2.1` which includes the auth fixes, so you can just use that

**Comment by dragon-D** [comment]
> @dragon-D can you collect a report via wireshark or tcpdump? It would be interesting to see which requests caused the error. Also, please rerun your scenario on the main branch, where the authentication problem is already fixed? (By specifying git as the dependency source in Cargo.toml: `scylla = { git = "https://github.com/scylladb/scylla-rust-driver" }`)

It may be related to my local network, I put it on the server and it does not appear at present
